core-web/apps/dotcms-ui/src/app/portlets/dot-categories/dot-categories-table/dot-categories-table.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component } from '@angular/core';
+import { DotCategory } from '@dotcms/app/shared/models/dot-categories/dot-categories.model';
+import { Observable } from 'rxjs';
+import {
+    DotCategoriesTableStore,
+    DotCategoriescTableState
+} from './store/dot-categories-table.store';
+
+@Component({
+    selector: 'dot-categories-table',
+    templateUrl: './dot-categories-table.component.html',
+    styleUrls: ['./dot-categories-table.component.scss'],
+    providers: [DotCategoriesTableStore]
+})
+export class DotCategoriesTableComponent {
+    vm$: Observable<DotCategoriescTableState> = this.store.vm$;
+    selectedCategories: DotCategory[] = [];
+    constructor(private store: DotCategoriesTableStore) {}
+
+    /**
+     * The function takes the value of the input field and passes it to the store
+     * @param event - The event object that was triggered by the user.
+     */
+    onChange(event) {
```

Comment:
Please use https://marketplace.visualstudio.com/items?itemName=oouo-diogo-perdigao.docthis for the documentation

Add types.